rubert/src/bin/validate.rs

```diff
@@ -0,0 +1,356 @@
+                    .into()
+                }
+
+                pipeline
+            }
+            ModelKind::TractSMBert => {
+                let pipeline = BertBuilder::from_files(model.vocab, model.model)
+                    .unwrap()
+                    .with_accents(tokenizer.accents)
+                    .with_lowercase(tokenizer.lowercase)
+                    .with_token_size(tokenizer.token_size)
+                    .unwrap()
+                    .with_pooling(NonePooler)
+                    .build()
+                    .unwrap();
+
+                Self::TractSMBert(pipeline)
+            }
+            ModelKind::TractQAMBert => {
+                let pipeline = BertBuilder::from_files(model.vocab, model.model)
+                    .unwrap()
+                    .with_accents(tokenizer.accents)
+                    .with_lowercase(tokenizer.lowercase)
+                    .with_token_size(tokenizer.token_size)
+                    .unwrap()
+                    .with_pooling(NonePooler)
+                    .build()
+                    .unwrap();
+
+                Self::TractQAMBert(pipeline)
+            }
+        }
+    }
+
+    /// Runs the model pipeline to infer the embedding of a sequence.
+    fn run(&mut self, sequence: impl AsRef<str>) -> Embedding2 {
+        match self {
+            Self::OnnxMBert {
+                tokenizer, session, ..
+            } => {
+                let encoding = tokenizer.encode(sequence);
+                let (token_ids, type_ids, _, _, _, _, attention_mask, _) = encoding.into();
+                let inputs = vec![
+                    Array1::<i64>::from(token_ids).insert_axis(Axis(0)),
+                    Array1::<i64>::from(attention_mask).insert_axis(Axis(0)),
+                    Array1::<i64>::from(type_ids).insert_axis(Axis(0)),
+                ];
+                let outputs = session.as_mut().unwrap().run(inputs).unwrap();
+
+                outputs[0].slice(s![0, .., ..]).to_owned().into()
+            }
+            Self::TractSMBert(pipeline) => pipeline.run(sequence).unwrap(),
+            Self::TractQAMBert(pipeline) => pipeline.run(sequence).unwrap(),
+        }
+    }
+}
+
+/// A validator to compare two models based on a set of Ted talks.
+struct Validator {
+    talks: &'static str,
+    skip: usize,
+    take: usize,
+    source: Pipeline,
+    target: Pipeline,
+    errors: Array1<f32>,
+}
+
+impl Validator {
+    /// Builds a validator from a configuration.
+    fn build<R: RangeBounds<usize>>(config: &ValidatorConfig<R>) -> Self {
+        let talks = config.data.talks;
+        let skip = match config.data.range.start_bound() {
+            Bound::Included(start) => *start,
+            Bound::Excluded(start) => start + 1,
+            Bound::Unbounded => 0,
+        };
+        let take = match config.data.range.end_bound() {
+            Bound::Included(end) => end + 1,
+            Bound::Excluded(end) => *end,
+            Bound::Unbounded => 2467,
```

Comment:
Shouldn't this be the number of lines in `talks`?

Reply:
interestingly, the number of lines in the csv file don't correspond to the number of records, probably due to some additional line breaks because of extra long talks.